src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -618,14 +618,14 @@ impl Tree {
 impl<'tcx> Tree {
     /// Insert a new tag in the tree.
     ///
-    /// `initial_perms` defines the initial permissions for the part of memory
-    /// that is already considered "initialized" immediately. The ranges in this
-    /// map are relative to `base_offset`.
+    /// `initial_perms` defines the initial permissions for a block of memory starting at
+    /// `base_offset`. These may nor may not be already marked as "accessed".
     /// `default_perm` defines the initial permission for the rest of the allocation.
+    /// These are definitely not "accessed".
     ///
     /// For all non-accessed locations in the RangeMap (those that haven't had an
     /// implicit read), their SIFA must be weaker than or as weak as the SIFA of
-    /// `default_perm`.
+    /// `default_perm`; otherwise, this will panic.
```

Comment:
@JoJoDeveloping  I am actually a bit confused by this. The assertion looks as follows:
```rust
                assert!(
                    default_strongest_idempotent
                        >= perm.permission.strongest_idempotent_foreign_access(protected)
                );
```
This seems to run for all locations in `initial_perms`, not just the non-accessed ones. Why that?

Reply:
It could be changed to the following, I think:
```
                assert!(
                    perm.is_accessed()
                        || default_strongest_idempotent
                            >= perm.permission.strongest_idempotent_foreign_access(protected)
                );
```
But the way it is currently makes it easier to reason about because the SIFA correction can be proven correct without considering the initial read.

Reply:
Then we should update the comments to match, though.

Reply:
Instead of tracking this assertion all the way through, wouldn't it be easier to just compute the maximum SIFA and pass that to `update_last_accessed_after_retag`? I've done that now in the 2nd commit, could you take a look if that makes sense?